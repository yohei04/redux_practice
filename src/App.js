import React from "react";
import { Provider } from "react-redux";
// import Posts from "./components/Posts";
// import PostForm from "./components/PostForm";
// import store from './store'
import Color from "./DE/components/Color";

import { useSelector, useDispatch } from "react-redux";
import { increment, decrement, multiplication } from "./DE/actions";

function App() {
  const counter = useSelector((state) => state.counter);
  const isLogged = useSelector((state) => state.isLogged);
  const dispatch = useDispatch();

  return (
    <div className="App">
      <Color />
      <h1>Counter {counter}</h1>
      <button onClick={() => dispatch(increment(5))}>+</button>
      <button onClick={() => dispatch(decrement())}>-</button>
      <button onClick={() => dispatch(multiplication(2))}>x</button>

      {isLogged ? <h3>Valuable I shouldn't see</h3> : ""}
    </div>
  );
}

export default App;
