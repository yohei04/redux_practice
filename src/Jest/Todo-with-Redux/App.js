import React from 'react'
import AddTodo from './components/AddTodo'
import TodoList from './components/TodoList';

const App = () => {
  return (
    <div>
      <AddTodo />
      <TodoList />
    </div>
  )
}

export default App

