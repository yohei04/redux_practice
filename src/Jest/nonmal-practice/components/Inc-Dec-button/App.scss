.button-section {
  color: red;
  text-align: center;
  font-size: 2rem;
}
