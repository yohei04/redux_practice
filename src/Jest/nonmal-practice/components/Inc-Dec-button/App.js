import React from 'react';
import Button from './Button';
import './App.scss';

const App = () => {
  return (
    <div>
      <Button />
    </div>
  );
};

export default App;
