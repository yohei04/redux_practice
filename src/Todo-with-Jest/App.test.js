import React from 'react';
import { render, cleanup } from '@testing-library/react';
import App from '../Jest/components/Inc-Dec-button/App';

it('should take a snapshot', () => {
  const { asFragment } = render(<App />);

  expect(asFragment(<App />)).toMatchSnapshot();
});
