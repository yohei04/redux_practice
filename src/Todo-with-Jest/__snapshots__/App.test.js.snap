// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should take a snapshot 1`] = `
<DocumentFragment>
  <div>
    <div
      class="button-section"
      data-testid="count"
    >
      <button>
        +
      </button>
      <span>
        0
      </span>
      <button>
        -
      </button>
    </div>
  </div>
</DocumentFragment>
`;
